epan/dissectors/packet-ippusb.c

```diff
@@ -21,70 +21,421 @@
+pdu_store(packet_info *pinfo, wmem_tree_t *multisegment_pdus)
+{
+    struct ippusb_multisegment_pdu *msp;
+
+    msp=wmem_new(wmem_file_scope(), struct ippusb_multisegment_pdu);
+    msp->first_frame=pinfo->num;
+    msp->last_frame=pinfo->num;
+    msp->finished = FALSE;
+    msp->flags=0;
+    wmem_tree_insert32(multisegment_pdus, pinfo->num, (void *)msp);
+
+    return msp;
+}
+
+struct ippusb_analysis {
+
+	wmem_tree_t *multisegment_pdus;
+
+};
+
+static struct ippusb_analysis *
+init_ippusb_conversation_data()
+{
+    struct ippusb_analysis *ippusbd;
+
+    /* Initialize the tcp protocol data structure to add to the tcp conversation */
+    ippusbd=wmem_new0(wmem_file_scope(), struct ippusb_analysis);
+
+    ippusbd->multisegment_pdus=wmem_tree_new(wmem_file_scope());
+
+    return ippusbd;
+}
+
+struct ippusb_analysis *
+get_ippusb_conversation_data(conversation_t *conv, packet_info *pinfo)
+{
+    struct ippusb_analysis *ippusbd;
+
+    if(conv==NULL ) {
+        conv = find_or_create_conversation(pinfo);
+    }
+
+    ippusbd=(struct ippusb_analysis *)conversation_get_proto_data(conv, proto_ippusb);
+
+    if (!ippusbd) {
+        ippusbd = init_ippusb_conversation_data();
+        conversation_add_proto_data(conv, proto_ippusb, ippusbd);
+    }
+
+    if (!ippusbd) {
+      return NULL;
+    }
+
+    return ippusbd;
+}
+
+
+static gpointer ippusb_temporary_key(const packet_info *pinfo _U_, const guint32 id _U_, const void *data)
+{
+    return (gpointer)data;
+}
+
+static gpointer ippusb_persistent_key(const packet_info *pinfo _U_, const guint32 id _U_, const void *data)
+{
+    return (gpointer)data;
+}
+
+static void ippusb_free_temporary_key(gpointer ptr _U_) { }
+
+static void ippusb_free_persistent_key(gpointer ptr _U_) { }
+
+reassembly_table_functions ippusb_reassembly_table_functions =
+{
+    g_direct_hash,
+    g_direct_equal,
+    ippusb_temporary_key,
+    ippusb_persistent_key,
+    ippusb_free_temporary_key,
+    ippusb_free_persistent_key
+};
+
 static dissector_table_t ippusb_dissector_table;
+static reassembly_table ippusb_reassembly_table;
 
+/* Main dissector function */
 static int
 dissect_ippusb(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, void* data)
 {
-    //skip to http header info
-    //put dummy info in for http dissector
-
-    proto_tree  *ippusb_tree = tree;
-    //proto_item  *ti;
-    int         offset     = 0;
-    //http_message_info_t *message_info = (http_message_info_t *)data;
-    //gboolean    is_request;
-    //guint16     operation_status;
-   // const gchar *status_type;
-    //guint32	request_id;
-    //conversation_t *conversation;
-
-    //offset +=
-
-    if (tvb_captured_length_remaining(tvb, offset) > 0) {
-        next_tvb = tvb_new_subset_remaining(tvb, offset);
-        offset += try_dissect_next_protocol(parent, next_tvb, pinfo, usb_conv_info, urb_type, tree, NULL);
+
+    gint reported_length = tvb_reported_length(tvb);
+    if (tvb_reported_length(tvb) <= 8) {
```

Comment:
>8 [](start = 36, length = 1)

Code comment for what 8 represents (or use a static const instead, set to 8 to document what this means)